en/ros2/px4_sdk.md

```diff
@@ -0,0 +1,336 @@
+- This allows multiple executors to coexist.
+- Executors cannot activate other executors.
+- Within the SDK, a mode executor is always implemented in combination with a custom mode.
+
+:::note
+- These definitions guarantee that a user can take away control from a custom mode or executor at any point in time by commanding a mode switch through RC or a GCS.
+- A mode executor is transparent to the user.
+  It gets indirectly selected and activated through the owning mode, and thus the mode should be named accordingly.
+:::
+
+##### When would you use a mode executor?
+
+The most common use-case is the execution of an autonomous 'mission', such as a survey or a payload delivery: the executor triggers a take-off, then switches to a custom mode which performs the survey, and when completed, triggers an RTL.
+
+#### Configuration Overrides
+
+Both modes and executors can define configuration overrides, allowing to customize certain behaviors while the mode or executor is active.
+
+These are currently implemented:
+
+- Disabling auto-disarm. This allows for landing and then taking off again (e.g. to release a payload).
+- Ability to defer non-essential failsafes.
+  It allows for example an executor to run a critical action without being interrupted by low-battery failsafe, such as controlling a winch.
+
+### Comparison to Offboard Control
+
+The above concepts provide a number of advantages over traditional [offboard control](../ros/offboard_control.md):
+
+- Multiple nodes or applications can coexist and even run at the same time. It is clearly defined which node can control the vehicle at any given time.
+- Exposing a selectable mode to the GCS.
+- Integrated with the failsafe state machine and arming checks.
+- It is well-defined which setpoint types can be sent.
+- It is possible to replace an FMU-internal mode (such as RTL).
+
+## Getting Started
+
+The following steps are required to get started:
+
+1. Make sure you have a working [ROS 2 setup](../ros/ros2_comm.md), with _px4_msgs_ in the ROS 2 workspace.
+2. Clone the SDK into the workspace:
+
+   ```shell
+   cd $ros_workspace/src
+   git clone --recursive https://github.com/PX4/px4_sdk.git
+   ```
+
+   :::note
+   To ensure compatibility, use the latest _main_ branches for PX4, _px4_msgs_ and the SDK.
+   See also [here](https://github.com/PX4/px4_sdk#compatibility-with-px4).
+   :::
+
+3. Build the workspace:
+
+   ```shell
+   cd ..
+   source install/setup.bash
+   colcon build
+   ```
+
+4. In a different shell, start PX4 SITL (you can use any model or simulator):
+
+   ```shell
+   cd $px4-autopilot
+   make px4_sitl gazebo-classic
+   ```
+
+5. Run the micro XRCE agent in a new shell (you can keep it running afterward):
+
+   ```shell
+   MicroXRCEAgent udp4 -p 8888
+   ```
+
+6. Start QGroundControl.
+
+   :::note
+   Use QGroundControl Daily, which supports dynamically updating the list of modes.
+   :::
+
+7. Back in the ROS 2 terminal, run one of the example modes:
```

Comment:
This is the reason I was thinking of the split  "Installing the SDK" and "Testing the Installation". This is something you do once to verify the installation - it's not instructions you do every time. On the other hand, it does help you understand how you might test installation.

What debug do you get by default and what have you injected in your ROS node? I.e.